web/pom.xml

```diff
@@ -922,10 +891,146 @@
     </plugins>
   </build>
 
-  <!-- Define profiles for running configuration using -Denv=profile.id parameter
-    Profiles configuration is defined in src/main/filters.
-    -->
   <profiles>
+
+    <!-- Unpack schema resources from maven artifacts -->
+    <profile>
+      <id>schemas-unpack</id>
+      <activation>
+        <property><name>schemasCopy</name><value>!true</value></property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-dependency-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>unpack-schemas</id>
+                <phase>process-resources</phase>
+                <goals><goal>unpack</goal></goals>
+                <configuration>
+                  <encoding>UTF-8</encoding>
+                  <artifactItems>
+                    <!-- unpack built-in schema plugins into schema-plugins directory -->
+                    <artifactItem>
+                      <groupId>org.geonetwork-opensource.schemas</groupId>
+                      <artifactId>schema-iso19139</artifactId>
+                      <type>zip</type>
+                      <overWrite>false</overWrite>
```

Comment:
@jodygarnett, questions you may have hints on.
* when a file is removed from a plugin, it is not removed from the data dir. I tried overWrite true without success. BTW overwrite helps when a file is removed from the datadir (which should not happen often) but when switching from one branch to another, you may end up with the wrong version?
* We used to be able to do `mvn process-resources` to update the plugin in the datadir - how do you do know? We have to build the plugin first and then process-resource for unpack or do you have a one step command?

Thanks

Reply:
removed files:

* I agree it would be smart to run "mvn clean" when switching branches.
* Reading [overright rules](https://maven.apache.org/plugins/maven-dependency-plugin/usage.html#Overwrite_Rules) is interesting but shows no way to have files removed

Correct two steps are required using this approach, but:

* You can still use `mvn process-resources -PcopySchema=true` if you want a one step command

* See next proposal[(https://github.com/geonetwork/core-geonetwork/wiki/Resolve-competing-web-module-war-definitions) for a long term solution...

Reply:
> * You can still use `mvn process-resources -PschemaCopy=true` if you want a one step command

Looks to not be working - maybe I'm doing something wrong? Ended up using https://github.com/geonetwork/core-geonetwork/pull/4946 which is faster.

Reply:
I have a proposal to make things easier long term for your consideration: https://github.com/geonetwork/core-geonetwork/wiki/Resolve-competing-web-module-war-definitions

Can I ask what "looks not to be working" means, do you have a concrete example to test?  The existing setup has schema plugins spliced into the war in multiple locations (and directly watched also). Were you able to see your change copied into the correct location? Was it still not recognized?

Reply:
Test case: 


- With GeoNetwork running with `mvn jetty:run`, in the schemas project, change a file, for example: `SOURCES/schemas/iso19139/src/main/plugin/iso19139/layout/layout.xsl`

Adding for example this following line `<xsl:message>TEST</xsl:message>`, below:

https://github.com/geonetwork/core-geonetwork/blob/master/schemas/iso19139/src/main/plugin/iso19139/layout/layout.xsl, adding a message in https://github.com/geonetwork/core-geonetwork/blob/57066481db0c4bd02b9e4c27e1784f6a5e0e2f23/schemas/iso19139/src/main/plugin/iso19139/layout/layout.xsl#L65-L67


- From the `web` module, execute `mvn process-resources -PschemaCopy=true``

- Check the file deployed: `SOURCES/web/src/main/webapp/WEB-INF/data/config/schema_plugins/iso19139/layout/layout.xsl`, doesn't contain the previous change.

Reply:
Please use `-DschemasCopy=true` command line option

Reply:
I tested locally by changing the following to verbose:

```
                  <target description="copy from schemas to schema_plugins folder">
                    <copy todir="${schema-plugins.dir}" preservelastmodified="true" verbose="true">
                      <fileset dir="${basedir}/../schemas/">
                        <include name="**/src/main/plugin/**"/>
                      </fileset>
                      <regexpmapper handledirsep="yes"
                        from="^[-_\.a-zA-Z0-9]+/src/main/plugin/(.*)"
                          to="\1" />
                    </copy>
```

And using `mvn process-resources -DschemasCopy=true`, it shows:

```
[INFO] --- maven-antrun-plugin:3.0.0:run (copy-schemas-ant) @ web-app ---
[INFO] Executing tasks
[INFO]      [copy] Copying 1 file to /Users/jgarnett/java/geonetwork/core-geonetwork/web/src/main/webapp/WEB-INF/data/config/schema_plugins
[INFO]      [copy] Copying /Users/jgarnett/java/geonetwork/core-geonetwork/schemas/iso19139/src/main/plugin/iso19139/layout/layout.xsl to /Users/jgarnett/java/geonetwork/core-geonetwork/web/src/main/webapp/WEB-INF/data/config/schema_plugins/iso19139/layout/layout.xsl
[INFO] Executed tasks
```

Reply:
The documentation appears correct in web/README.md, I am sorry it appears I had the `-D` example incorrect in this thread earlier and confused everyone. I am much more focused on making sure the docs are correct.

Reply:
@josegar74 I like the use of `preservelastmodified="true"` above, perhaps you can add that to `master`.

Reply:
Here is a small PR for that https://github.com/geonetwork/core-geonetwork/pull/4948